routes/quiz.js

```diff
@@ -0,0 +1,116 @@
+const express = require('express');
+const { updateNotification } = require('npm');
+const db = require('../db');
+const router = express.Router()
+
+
+//Get all quiz data
+router.get('/', (req, res) => {
```

Comment:
If I navigate directly here, I don't get challenged even if I'm not logged in.
I think we want to add an explicit check here, e.g.:
```suggestion
router.get('/', (req, res) => {
    if (!req.user) {
        res.redirect('/')
    }

```

Reply:
And if we wanted to handle perms, we'd have to consider them with something like:
```js
router.get('/:id/create', (req, res) => {
    db.get("SELECT * FROM users WHERE username = $username", {
        $username: req.user.username
    }, (err, user) => {
        if (err) {
            // handle that case, maybe redirect them to home page or show an error
        }
        if (!permissions.get(actions.VIEW_FILE).includes(user.role_id)){
            console.log("invalid auth")
            // handle that case
        }
       // let them through
```